### ISSUE TITLE
Discussion: Anti-Phishing Protection

### ISSUE THREAD

**Issue by hhpmgr** [gophish/gophish#977]
Looking for some knowledge-sharing (please let me know if this is the wrong place for this discussion, or if it already exists, I couldn't find it...)

Trying to run an authorised phishing assessment against an organisation is one thing, but what about dealing with anti-phishing measures in place from providers such as dropbox and google? Some login pages are getting more sophisticated with protections in place to make cloning pages harder, and identifying suspicious activity swifter. For instance in the process of creating a dropbox phishing template, I saw one test google  chrome browser start to throw the red warning page within a few hours. This makes me think that dropbox is sharing information with google, or google is getting smarter at auto-detecting warning signs, whether it be new domains, non-HTTPS pages with login forms, and suspicious domain names, or some combination of the above.

So, I'm looking for some thoughts on how to run an authorised phishing assessment that clones third party services.... or is this just a bad idea? I appreciate it is a fine line between this and real malicious activity...

Thanks in advance.

**Comment by S0larflare** [comment]
I wouldn't be surprised if some of the larger sites, those which are prime targets for phishing: apple, google, paypal, dropbox etc. have scripts which automatically flag domains through an api if their scripts are called from anywhere but their home domain. It would be pretty effective if a site is cloned and the scripts aren't removed. That's pretty baseless though to be honest, so take it with a pinch of salt.

The best way I find is to clone a page and then rework it. Many of the 'big' sites don't clone perfectly anyway so as part of 'fixing' the page I try to remove any scripts and references back to the original domain. Admittedly I do my campaigns internally so Google et. al. couldn't do much about it anyway, but it is good practice to clean things up.

I'm not sure how the 'professional' phish testing companies do it (PhishMe, KnowBe4 etc.), but I would have thought they would work in the same sort of way, have a page that doesn't call back home at all. I'd imagine they may have some mechanism for whitelisting it with the owner of the domain too though.

**Comment by hhpmgr** [comment]
Thanks @S0larflare I was thinking along the same lines. So: creating phishing landing pages is potentially pretty tricky. I see that people are a little cagey about sharing templates for this, and possibly for good reason... Are you aware of any resources? I have taken to cloning an actual phishing page used against my org, as this had a lot of the grunt work done already, but obviously this will come with some risks. The one I used was fairly simple, but it's quite possible that a malicious phisher could include their own sneaky js that could slip under the radar and send creds back to their own servers etc....

Is the best (or only real) option to get down and dirty and build landing pages from the ground up?

**Comment by rfdevere** [comment]
Tip - Don't rip off a legit PayPal, Google etc.... go find realistic real life phishing pages, clone them.... 
One massive consideration is sanitising the filthy Java script nasties but if you strip it back, 9/10 it's a pretty smooth template. Maybe search Twitter for 'hxxp' to start you off. 

As for evasion, again... i'm not advocating crime at all but they do have the best techniques. In educational phishing please don't waste your time - whitelist so you can reach more inboxes, train more people. If you want to test your organisation in a realistic manner, spear phish style I wouldn't even use Gophish as everybody looking for the servers in cloud based email gateway world anyway. If you want to avoid the red page you phish discreetly or whitelisted.

**Comment by hhpmgr** [comment]
Thanks @rfdevere that's the approach that has been most useful so far in cloning attacks used against my org. I'll try searching twitter as well :-) 

We do whitelist on all our incoming services, as the point isn't to test our technical defences, it's to test the human in the cases when technology fails 👍 

Spear-phishing I've not tried yet but will be looking to simulate this as well this year.

Thanks

**Comment by rfdevere** [comment]
This is me https://theantisocialengineer.com/professional-services/phishing/ @antiSocial_Eng on Twitters. If you need few tips shout me.

**Comment by hhpmgr** [comment]
Thanks, will do. hxxp is a bit of a goldmine of nasties! But need to be fast as they tend to get taken down pretty quick.

**Comment by jordan-wright** [comment]
These discussions are always good - thanks for sending this over @hhpmgr. I agree with everyone's advice here. At a high level:

* If you leave cloned pages as-is, they will get taken down quickly. **This is a good thing!** I understand how it can be frustrating when wanting to run a legitimate campaign, but it's fantastic that service providers are so quick to shutdown what could otherwise be malicious page.

* Instead, consider altering the pages. I personally highly recommend removing any branding elements like logos and replace them with generic alternatives. This shouldn't affect the quality of the test and gives the benefit that your helping protect the established brands when it comes to your users' perspective.

* If you're looking for inspiration, the hxxp filter is good. You can always refer to community curated sites like [Phishtank](https://www.phishtank.com/) and [OpenPhish](https://openphish.com/) for inspiration. As always, be careful when interacting with any of those sites/samples.

* Whitelisting is a good thing. If you're running a legitimate test, it's always encouraged to whitelist your sender. You won't be able to whitelist your site from things like Chrome's SafeBrowsing (again, a good thing!) but you can at least ensure that your emails are received.

**Comment by hhpmgr** [comment]
Here is a nice example of why you ought to be careful when cloning real phishing sites. So, openphish is a great resource, and this site popped up as a dropbox phishing site: hxxp://infotechmedia[.]com/sx/s/ .

I cloned it first with wget -p -k, then used sed -i to replace any links that were missed in the rewrite. Great, a working clone page (for me to then clone with Gophish). In incognito etc. my AV blocked attempts to hit my cloned site, so I thought either I missed an included resource, or there is some nasty javascript. Looking at the javascript (which was now being served by my web server), and there was one script titled "new wave javascript" being included in the page. This was doing some fairly heavy well obfuscated rewriting of itself using variables and regex. In a safe environment I cloned the page with gophish and ran a test campaign. The data collected included my test email and pw, plus a variable that did not show up at all in the HTML source: "server":"jameswds6@gmail.com". So my assumption is currently that any data input would be sent somehow through the javascript directly to that email address.

Learning about this all the time but I guess the main takeaway here is : be very careful. The javascript could have been doing something even more nasty.

It would be great if there was a resource somewhere of "safe" phishing templates. I understand there would be some challenges in getting that up and running.
